sematic/ui/src/components/RunPanel.tsx

```diff
@@ -104,9 +104,23 @@ export default function RunPanel(props: {
             <Box sx={{ paddingBottom: 3, gridColumn: 1 }}>
               <Box marginBottom={3}>
                 <Typography variant="h6">{selectedRun.name}</Typography>
```

Comment:
> I think what would be even more useful is a link to the original run.
> For example, instead of printing the entire Original Run ID. How about:
> ```
> Run ID: <run-ID> (cloned from <link><trimmed-original-run-ID></link>)
> ```
> You do a trimmed run ID with:
> ```typescript
> <RunId runId={run.original_root_id} trim />
> ```

@neutralino Currently the deep linking to non-root runs doesn't work that well, but I implemented your suggestion, given than we'll get around to fixing this anyway, and that this feature doesn't manifest on the usual happy path.
cc @chance-an 

![image](https://user-images.githubusercontent.com/1894533/207048042-ce1d62eb-4999-4644-8356-a37b5e723234.png)

![image](https://user-images.githubusercontent.com/1894533/207048418-d4cc7ac2-49ea-4b9d-bf4a-45b3785aecc4.png)

Reply:
It looks like we are not quite ready to deep-link nested runs. Let's wait for that to happen before we implement this change. In order words, let's withdraw this change of showing original ID for now.

In the meantime, we also need the root id of the nested run to use the deep link of nested runs. Do we have a way to find that easily? Maybe take a look at how this is done to get prepared beforehand.

Reply:
@chance-an the root ID for the current nested run is available on the current nested run's `Run` model object. What we don't have is the root ID of the original run from which the current run is cached/cloned. THis can be easily fetched by querying the API for the original run.

Two things we could do regarding deep-linking:
- `https://address/runs/<root-run-ID>/<nested-run-ID>` and display the appropriate info
- `https://address/runs/<nested-run-ID>` and we automatically figure out the root run ID to fetch. We always fetch the entire graph anyway.

The second option is neater as a shorter URL.

Reply:
Thanks for the input. 

I think `https://address/runs/<root-run-ID>/<nested-run-ID>` is probably more economical to implement because it fits the user's most common usage pattern --- in that they always go to the root run first, then navigate to a nested run. One key focus of deep-linking is to recover a previous UI state user has developed into through a series of interactions. The second pattern, `https://address/runs/<nested-run-ID>`, works against this principle by eliminating some important information on the URL.  It sure does make it possible to achieve the same thing. But anything works against the semantics usually incur unnecessary complexity in code.

If we overload the semantics of `https://address/runs/<root-run-ID>/***`  and `https://address/runs/<nested-run-ID>`, it confuses the FE routing logic to some degree as well. 

Taking a step back, if we desire the second form, `https://address/runs/<nested-run-ID>`, a redirection to the first form might involve the least complexity. This means when we see `https://address/runs/<nested-run-ID>` and know that the run ID is indeed a nested run, then find the root ID of the run, then redirect to render `https://address/runs/<root-run-ID>/<nested-run-ID>` instead. We always use the first form as the primary basis.

Reply:
separately, I have seen some products support **short-url** mechanism baked in. 

Eg. `http:\\webviz.cruise.com\view\[view_id]` would eventually expand to `http:\\webviz.cruise.com\some\business\and\a\lot\other\parameters`. These settings can be serialized within the BFF (back for frontend), without the knowledge of the backend API server. So the long URLs problem can also be solved this way (in the long term).

Reply:
I'm thinking we should design this feature to help us devs debug things easily. In that regard, the `https://address/runs/<nested-run-ID>` API is the easiest to use. Given a Run ID that we obtain from anywhere (the UI, the DB, the log, a scratch note), we can immediately construct the URL to get info about that Run, without having to consult a cheatsheet.

The Run data model itself contains the Resolution ID, so it is trivial to source both the Run panel and the Resolution panel. We can even do it via a single custom query with a join, without requiring 2 DB calls.

Short links like `http:\\webviz.cruise.com\view\[view_id]` are useful when you want to share links without taking too much space. Their contents isn't human readable. I'm not convinced we need this feature.

Reply:
Thanks for sharing this use case. It makes sense. We can also support that by doing a redirection (like the way I explained above in the `Taking a step back`) 

If we don't stick with the first form as the basis, I envision we might run into some issues in the following scenario:

If a user first land on `https://address/runs/<root-run-ID>/,` then click on a nested run. Since the user might later want to copy/paste the URL from the address bar, we must reflect this user action onto the URL. Otherwise, we miss a critical goal of deep-linking --- recovering the previous UI state. Then the question is, shall we change the URL to `https://address/runs/<root-run-ID>/<nested-run-ID>` or `https://address/runs/<nested-run-ID>`?  

Doing the second form will likely confuse the React router and incur an unnecessary full-page re-rendering. It is worth emphasizing that this is a series of user actions from a single page. It is different than pasting `https://address/runs/<nested-run-ID>` into the address bar and loading the page. So causing full-page reloading is terrible. 

By leveraging the first form, not only do we keep the user actions remain on the same page (same routing rule) we can still maintain some additional view parameters like highlighted tabs, etc. It is harder to do with the second form. Because the root-run information is gone, all the user actions coming from the root-run are also lost. We can't keep everything logically coherent since the series of user actions were not developed using the `https://address/runs/<nested-run-ID>` form.


Therefore, I guess a redirection approach could make two worlds happy. Simply put, this means that we should implement both.